Sources/SwiftParser/Types.swift

```diff
@@ -1052,7 +1052,35 @@ extension Parser {
         )
       )
     } else {
-      return self.parseType()
+      var result = self.parseType()
+
+      guard !result.hasError else {
+        return result
+      }
+
+      if self.at(.rightSquareBracket) {
+        let (unexpectedBeforeRSquareBracket, rightSquareBracket) = self.expect(.rightSquareBracket)
+        result = RawTypeSyntax(
+          RawArrayTypeSyntax(
+            leftSquareBracket: missingToken(.leftSquareBracket),
+            elementType: result,
+            unexpectedBeforeRSquareBracket,
+            rightSquareBracket: rightSquareBracket,
+            arena: self.arena
+          )
+        )
+      }
+
+      if !self.currentToken.isAtStartOfLine {
+        if self.at(.postfixQuestionMark) {
+          result = RawTypeSyntax(self.parseOptionalType(result))
+        }
+        if self.at(.exclamationMark) {
+          result = RawTypeSyntax(self.parseImplicitlyUnwrappedOptionalType(result))
+        }
+      }
```

Comment:
Should this be a loops so that we also parse two trailing question marks?

Also, I would prefer to use `self.at(TokenSpec(.postfixQuestionMark, allowAtStartOfLine: false))` instead of the nested `if` statements because it works better with the fuzzing I want to introduce in https://github.com/apple/swift-syntax/pull/1340.

Reply:
Not sure I understand?
Can you give an example so I also can add it as a test case.

Reply:
Added.

Not sure if the name of the while loop is good enough 🫣